docs/spec/concepts.rst

```diff
@@ -3,6 +3,272 @@
 Type system concepts
 ====================
 
+Static, dynamic, and gradual typing
+-----------------------------------
+
+A **statically typed** programming language runs a type checker before running
+a program. The program is required to be well typed according to the language's
+type system. The type system assigns a type to all expressions in the language
+and verifies that their uses obey the typing rules. Normally, a program that is
+not well typed (i.e., one that contains a type error) will not run. Java and
+C++ are examples of statically typed object-oriented languages.
+
+A **dynamically typed** programming language does not run a type checker before
+running a program. Instead, it checks the types of values before performing
+operations on them at runtime. This is not to say that the language is
+"untyped". Values at runtime have a type and their uses obey typing rules. Not
+every operation will be checked, but certain primitive operations in the
+language such as attribute access or arithmetic are. Python is a
+dynamically typed language.
+
+**Gradual typing** is a way to combine static and dynamic typing.
+Type-annotated Python allows opting in to static type checking at a fine level
+of granularity, so that some type errors can be caught statically, without
+running the program. Variables, parameters, and returns can optionally be given
+a static type annotation. Even within the type of a single data structure,
+static type checking is optional and granular. For example, a dictionary can be
+annotated to enable static checking of the key type but only have dynamic
+runtime checking of the value type.
+
+A **gradual** type system is one in which a special "unknown" or "dynamic" type
+is used to describe names or expressions whose types are not known statically.
+In Python, this type is spelled :ref:`Any`. Because :ref:`Any` indicates a
+statically unknown type, the static type checker can't check type correctness
+of operations on expressions typed as :ref:`Any`. These operations are still
+dynamically checked, via the Python runtime's usual dynamic checking.
+
+The Python type system also uses ``...`` within :ref:`Callable` types and
```

Comment:
The glossary defines the term "gradual form" for these. It may be useful to also use that term here, so you don't have to repeat the references to Callable and tuple in the next section.

Reply:
Oh yes, thank you, I intended to do that after adding the glossary entry, but then forgot to get back to it.